src/components/ble/NotificationManager.cpp

```diff
@@ -124,8 +124,8 @@ bool NotificationManager::AreNewNotificationsAvailable() const {
   return newNotification;
 }
 
-bool NotificationManager::ClearNewNotificationFlag() {
-  return newNotification.exchange(false);
+void NotificationManager::ClearNewNotificationFlag() {
+  newNotification = false;
```

Comment:
You remove the only use of atomic here. I didn't check for all sites where the function is called, but because it builds, I guess it is ok and the return value was never used.
So I am ok with changing this variable as well.